pretix_newsletter_manual/exporters.py

```diff
@@ -38,15 +37,19 @@ def iterate_list(self, form_data):
                 r.order.email,
                 r.created.astimezone(tz).strftime("%Y-%m-%d"),
             ]
-
-            try:
-                row += [
-                    r.order.invoice_address.name,
-                ]
+            if r.order.invoice_address.name:
+                row += [r.order.invoice_address.name]
                 if self.event and len(name_scheme["fields"]) > 1:
                     for k, label, w in name_scheme["fields"]:
                         row.append(r.order.invoice_address.name_parts.get(k, ""))
-            except InvoiceAddress.DoesNotExist:
+            elif r.order.positions.all()[0].attendee_name:
```

Comment:
This only works if the first item in cart has an attendee_name. I think it would be better to use the first item, that has a name with. e.g. something like `r.order.positions.filter(attendee_name_cached__isnull=False)[0]` (sorry, untested)

Reply:
This is also problematic for more reasons, even with the improvement. The way this is currently implemented, it will make **three** queries to the database **per line**, so it will overload the database with thousands of queries if there are thousands of lines!

Better: Preload all order positions in the queryset above, like this:

```
qs = NewsletterRequest.objects.filter(order__event__in=self.events).order_by(
    "created"
).prefetch_related('order', 'order__all_positions')
```

See also: https://docs.djangoproject.com/en/4.0/topics/db/optimization/#retrieve-everything-at-once-if-you-know-you-will-need-it

Then, for every row, filter on Python level:

```
possible_positions = [op for op in order.all_positions.all() if not op.canceled and op.attendee_name]
name_parts = possible_positions[0] if possible_positions else {}
```